cmd/initialize/prepare.go

```diff
@@ -0,0 +1,74 @@
+package initialize
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"numerous.com/cli/cmd/initialize/wizard"
+	"numerous.com/cli/cmd/output"
+	"numerous.com/cli/internal/dir"
+	"numerous.com/cli/internal/manifest"
+	"numerous.com/cli/internal/python"
+)
+
+func PrepareInit(args []string) (string, *manifest.Manifest, error) {
+	appDir, err := os.Getwd()
+	if err != nil {
+		slog.Info("An error occurred when trying to get the current user path during init process.", slog.String("error", err.Error()))
+		fmt.Println(err)
+
+		return "", nil, ErrGetWorkDir
+	}
+
+	if len(args) != 0 {
+		appDir = PathArgumentHandler(args[0], appDir)
+	}
+
+	if exist, _ := dir.AppIDExists(appDir); exist {
+		output.PrintError(
+			"An app is already initialized in \"%s\"",
+			"💡 You can initialize an app in another folder by specifying a\n"+
+				"   path in the command, like below:\n\n"+
+				"numerous init ./my-app-folder\n\n",
+			appDir,
+		)
+
+		return "", nil, ErrAppAlreadyInitialized
+	}
+
+	lib, err := manifest.GetLibraryByKey(libraryKey)
+	if libraryKey != "" && err != nil {
+		output.PrintErrorDetails("Unsupported library", err)
+		os.Exit(1)
```

Comment:
return, I think we should try to be more consistent in what level returns errors and what can use `os.Exit`

Reply:
Yes. I changed this to return an error instead.